From 98c6e45d6f9d6a59572c455b8760087d810a6862 Mon Sep 17 00:00:00 2001
From: danielhk <daniel.p6800@gmail.com>
Date: Sun, 3 Dec 2017 21:39:24 +0800
Subject: [PATCH 1/5] Add error message suppress option

Suppress the error message "E AudioALSAStreamOut:
-getPresentationPosition(), no playback handler, *frames = 0, return"
This is not really an error but repeated endlessly all the time.
---
 audio/2.0/default/Android.mk    | 4 ++++
 audio/2.0/default/StreamOut.cpp | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/audio/2.0/default/Android.mk b/audio/2.0/default/Android.mk
index 621853cb..84ca6e3c 100644
--- a/audio/2.0/default/Android.mk
+++ b/audio/2.0/default/Android.mk
@@ -88,4 +88,8 @@ ifeq ($(TARGET_USES_BCRADIO_FUTURE_FEATURES),true)
 LOCAL_CFLAGS += -DTARGET_USES_BCRADIO_FUTURE_FEATURES
 endif
 
+ifeq ($(SUPPRESS_MTK_AUDIO_BLOB_ERR_MSG),true)
+LOCAL_CFLAGS += -DSUPPRESS_MTK_AUDIO_BLOB_ERR_MSG
+endif
+
 include $(BUILD_EXECUTABLE)
diff --git a/audio/2.0/default/StreamOut.cpp b/audio/2.0/default/StreamOut.cpp
index 290d0b10..14214121 100644
--- a/audio/2.0/default/StreamOut.cpp
+++ b/audio/2.0/default/StreamOut.cpp
@@ -493,6 +493,9 @@ Result StreamOut::getPresentationPositionImpl(audio_stream_out_t* stream,
     // continuously querying it, but the stream has been stopped.
     static const std::vector<int> ignoredErrors{EINVAL, EAGAIN, ENODATA};
     Result retval(Result::NOT_SUPPORTED);
+#ifdef SUPPRESS_MTK_AUDIO_BLOB_ERR_MSG
+    if (true) return retval;
+#endif
     if (stream->get_presentation_position == NULL) return retval;
     struct timespec halTimeStamp;
     retval = Stream::analyzeStatus("get_presentation_position",
-- 
2.14.1

